pandas/core/ops/__init__.py

```diff
@@ -498,8 +498,17 @@ def column_op(a, b):
         # in which case we specifically want to operate row-by-row
         assert right.index.equals(left.columns)
 
-        def column_op(a, b):
-            return {i: func(a.iloc[:, i], b.iloc[i]) for i in range(len(a.columns))}
+        if right.dtype == "timedelta64[ns]":
+            # ensure we treat NaT values as the correct dtype
+            right = np.asarray(right)
```

Comment:
what if you *always* do ``right = np.asarray(right)``? (and just use a positional indexer in ``column_op``)

Reply:
That will be incorrect for EA dtypes

Reply:
(well, not incorrect, but potentially costly)

Reply:
Given that Jeff and I had the same question, I think that a comment here would be helpful.

Reply:
i like this heuristic, will do